controllers/relayer_controller.go

```diff
@@ -0,0 +1,224 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	terrav1alpha1 "github.com/terra-rebels/terra-operator/api/v1alpha1"
+)
+
+// RelayerReconciler reconciles a Relayer object
+type RelayerReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=terra.terra-rebels.org,resources=relayers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=terra.terra-rebels.org,resources=relayers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=terra.terra-rebels.org,resources=relayers/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Relayer object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *RelayerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Reconciling Relayer object")
+
+	relayer := &terrav1alpha1.Relayer{}
+	err := r.Client.Get(ctx, req.NamespacedName, relayer)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	pod := newPodForRelayer(relayer)
+
+	if err := controllerutil.SetControllerReference(relayer, pod, r.Scheme); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	foundPod := &corev1.Pod{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, foundPod)
+
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+
+		err = r.Client.Create(context.TODO(), pod)
+
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	} else if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *RelayerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&terrav1alpha1.Relayer{}).
+		Complete(r)
```

Comment:
Your missing some logic to register your "child pod". E.g @ https://github.com/classic-terra/terra-operator/blob/7fe8fa220b22b937b6767be1224400dd79467110/controllers/terradnode_controller.go#L44

If you dont do this k8s will not clean up the pod your controller creates when you delete the crd resource that is managed by the controller manager